ocaml/xcp-rrdd/scripts/rrdd/rrdd.py

```diff
@@ -223,7 +223,9 @@ def lazy_complete_init(self):
         self.path = self.dispatcher.get_path({"uid": self.uid})
         base_path = os.path.dirname(self.path)
         if not os.path.exists(base_path): os.makedirs(base_path)
-        self.dest = open(self.path, "wb")
+        # Open file at pos 0, creating but not truncating the file(not possible otherwise):
+        fd = os.open(self.path, os.O_RDWR|os.O_CREAT)
+        self.dest = os.fdopen(fd, mode='r+b')
```

Comment:
I started on a unit test just because it should be easy.

After I changed it to not mock fdopen, it revealed that Python2 `os.fdopen()` does not take keyword arguments, doh!
```py
>       self.dest = os.fdopen(fd, mode='r+b')
E       TypeError: fdopen() takes no keyword arguments
```
```suggestion
        self.dest = os.fdopen(fd, "r+b")
```